enclave/core/sgx/td.c

```diff
@@ -168,6 +168,30 @@ oe_sgx_td_t* oe_sgx_get_td()
     return td;
 }
 
+/*
+**==============================================================================
+**
+** oe_sgx_set_td_exception_handler_stack()
+**
+**     Internal API that allows an enclave to setup stack area for
+**     exception handlers to use.
+**
+**==============================================================================
+*/
+bool oe_sgx_set_td_exception_handler_stack(void* stack, uint64_t size)
+{
+    oe_sgx_td_t* td = oe_sgx_get_td();
+
+    /* ensure the stack + size is 16-byte aligned */
+    if (((uint64_t)stack + size) % 16)
+        return false;
+
+    td->exception_handler_stack_size = size;
```

Comment:
The compiler is free to reorder these assignments and it sometimes it will:
https://godbolt.org/z/EzE7bcd4Y

There needs to be a compiler barrier between the assignments to ensure ordering. Since, assignment is done to thread specific variables that are only used by the current thread, there is no need for an acquire-release pattern.

Reply:
addressed by checking if the size is zero